x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/optimizer/Optimizer.java

```diff
@@ -1090,7 +1091,21 @@ private Expression combine(And and) {
 
             boolean changed = false;
 
-            for (Expression ex : Predicates.splitAnd(and)) {
+            List<Expression> andExps = Predicates.splitAnd(and);
+            // Ranges need to show up before BinaryComparisons in list, to allow the latter be optimized away into a Range, if possible
+            andExps.sort(new Comparator<Expression>() {
```

Comment:
Maybe extract this custom Comparator into its own variable inside `CombineBinaryComparisons` and re-use that, without creating it each time the `combine` method is called?

Reply:
Good point, thanks.
I've however replaced with a (non-capturing) lambda, which should then be optimised by the jvm.